Update

DaniGTA opened the pull request:
# New Functions:
+ Hide SRV
+ Use Internal IP
+ IPv6
+ Custom get IP urls
+ Custom API url (i dont think that anyone will need it)
+ Zones
+ Last Update Timestamp
+ Show current changing IP
+ Force IP Change

# Updates
+ .NET Framework 4.0 => 4.6 #NeedForSSLConnection with 4.0 i cant connect too APIv4
+ Cloudflare API to v.4

DaniGTA committed:
Added: IPv6 Support

+ IPv6
+ Custom IP lookup URL for IPv6/IPv4
+ Update UI after IP changes

DaniGTA committed:
Update to Cloudflare API v4

+ Zone support

DaniGTA committed:
Bug fix

The Proxied (bool) value will not be reseted

DaniGTA committed:
Update README.md

DaniGTA committed:
Merge remote-tracking branch 'origin/master'

DaniGTA committed:
New Functions and bug fixes

+ Use internal ip
+ Hide SRV

- Removed crash bug with old configs

birkett commented on Classes/CloudFlareAPI.cs:
```diff
@@ -21,221 +21,357 @@
+        /// And return changes
         /// </summary>
-        public void updateRecords(JsonResponse fetchedRecords)
+        public List<Result> updateRecords(Get_dns_records_response fetchedRecords)
```
Class names should not contain underscores, please rename to GetDnsRecordsResponse

birkett commented on Classes/CloudFlareAPI.cs:
```diff
@@ -21,221 +21,357 @@
                 //Skip over MX and CNAME records
                 //TODO: Dont skip them :)
-                if (fetchedRecords.response.recs.objs[i].type != "A")
+                bool NeedIp = false;
```
This can be merged into a switch, with default case.
case "A": // fall through.
case: "AAAA":
   NeedIp = true;
default:
   NeedIp = false;

birkett commented on Classes/CloudFlareAPI.cs:
```diff
@@ -21,221 +21,357 @@
         /// </summary>
         /// <returns>JSON stream of records, null on error</returns>
-        public JsonResponse getCloudFlareRecords()
+        public Get_dns_records_response getCloudFlareRecords(string SelectedZone)
```
Avoid underscores in method names, recommend GetDnsRecordsResponse

birkett commented on Classes/JsonObjects/Cloudflare/Basic_response.cs:
```diff
@@ -0,0 +1,280 @@
+﻿using System;
```
Rename to BasicResponse.cs

birkett commented on Classes/JsonObjects/Cloudflare/Get_dns_records_response.cs:
```diff
@@ -0,0 +1,30 @@
+﻿namespace CloudFlareDDNS.Classes.JsonObjects.Cloudflare
```
Rename to GetDnsRecordsResponse.cs

birkett commented on Classes/JsonObjects/Cloudflare/Get_zone_list_response.cs:
```diff
@@ -0,0 +1,30 @@
+﻿namespace CloudFlareDDNS.Classes.JsonObjects.Cloudflare
```
Rename to GetZoneListResponse.cs

birkett commented on CloudFlareDDNS.csproj:
```diff
@@ -9,10 +9,11 @@
     <RootNamespace>CloudFlareDDNS</RootNamespace>
     <AssemblyName>CloudFlareDDNS</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.7</TargetFrameworkVersion>
```
I'd like to keep this on 4.0 if possible. Is there anything in here requiring 4.7?
XP last supported by 4.03
Vista last supported by 4.6

DaniGTA replied:
WebRequest cant connect too Cloudflare some ssl problems 4.7 just fix it, i can try a lower one

DaniGTA replied:
So i test it i cant get it lower then 4.6 it can maybe run in 4.0 with third party WebHandler

birkett commented on CloudFlareDDNS.csproj:
```diff
@@ -25,15 +26,14 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
+    <Optimize>true</Optimize>
```
Please leave optimisation disabled for debug builds.

birkett commented on CloudFlareDDNS.csproj:
```diff
@@ -61,7 +63,25 @@
   <PropertyGroup>
     <ApplicationIcon>Resources\AppIcon.ico</ApplicationIcon>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'DebugenTest|AnyCPU'">
```
What is the purpose of this new config?

DaniGTA replied:
was a test

birkett commented on CloudFlareDDNS.csproj:
```diff
@@ -61,7 +63,25 @@
+    <StartupObject>CloudFlareDDNS.Program</StartupObject>
+  </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```
Can this lib be avoided? Would prefer to use native .net serialization

DaniGTA replied:
yes will be removed

birkett commented on CloudFlareDDNS.sln:
```diff
@@ -1,22 +1,28 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.31101.0
+# Visual Studio 15
```
Anything in this PR need 2015? Recommend keeping on 2013 unless required.

birkett commented on Forms/frmMain.cs:
```diff
@@ -75,40 +93,79 @@ private void updateHostsList(JsonResponse fetchedRecords)
                 {
-                    row.Checked = true;
-                }
+                    case "A":
```
This can also be cleaned up, as above

birkett commented on Forms/frmMain.cs:
```diff
@@ -75,40 +93,79 @@ private void updateHostsList(JsonResponse fetchedRecords)
+                    listViewRecords.Items[j].ForeColor = System.Drawing.Color.Gray;
                 }
             }
-
```
I'm not against removing this whitespace, but:
1) If they are being removed, remove them everywhere.
2) This makes the PR bigger than it needs to be.

birkett commented on Forms/frmMain.cs:
```diff
@@ -345,32 +428,36 @@ private void autoUpdateTimer_Tick(object sender, EventArgs e)
-                this.Invoke(new addLogEntryInvoker(addLogEntry), row);
+                Logger.reset(); //Clear the log
+            }
+            catch (Exception)
```
What exceptions are being caught here? Would be better to handle them properly?

DaniGTA replied:
very rare Exception had it once so i take this solution, because i cant reproduce it.

birkett commented on Forms/frmSettings.Designer.cs:
```diff
@@ -63,115 +61,300 @@ private void InitializeComponent()
             this.lblEmailAddress.Size = new System.Drawing.Size(76, 13);
             this.lblEmailAddress.TabIndex = 4;
-            this.lblEmailAddress.Text = Properties.Resources.Settings_EmailAddress;
+            this.lblEmailAddress.Text = "Email Address:";
```
Please use globals (as above) or properties (as left) for these strings.
If anyone wants to translate the application, all strings need to be in one place.

birkett commented on Properties/AssemblyInfo.cs:
```diff
@@ -40,7 +40,7 @@
 // to COM components.  If you need to access a type in this assembly from 
 // COM, set the ComVisible attribute to true on that type.
-[assembly: ComVisible(false)]
+[assembly: ComVisible(true)]
```
Why?

birkett commented on Properties/AssemblyInfo.cs:
```diff
@@ -56,8 +56,8 @@
 
-[assembly: AssemblyVersion(Version.szBaseVersion)]
-[assembly: AssemblyFileVersion(Version.szBaseVersion)]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Please revert back to Version.* strings. This is important for the "official" builds.

birkett commented on Properties/GeneratedVersion.cs:
```diff
@@ -25,4 +25,4 @@
 ///    !!! THIS FILE IS GENERATED BY THE BUILD PROCESS. DO NOT EDIT !!!
 /// </summary>
-class GeneratedVersion { public const string szAdditionalVersion = "dev"; } 
+class GeneratedVersion { public const string szAdditionalVersion = "Unofficial"; } 
```
Please revert.

birkett commented on Properties/Resources.Designer.cs:
```diff
@@ -1,10 +1,10 @@
 // <auto-generated>
-//     This code was generated by a tool.
-//     Runtime Version:4.0.30319.34209
+//     Dieser Code wurde von einem Tool generiert.
```
English only please.

birkett commented on Properties/Resources.Designer.cs:
```diff
@@ -447,7 +447,7 @@ internal class Resources {
         
         /// <summary>
-        ///   Looks up a localized resource of type System.Drawing.Bitmap.
+        ///   Sucht eine lokalisierte Ressource vom Typ System.Drawing.Bitmap.
```
Looks like this whole file needs to be reverted.

birkett commented on Properties/Settings.Designer.cs:
```diff
@@ -0,0 +1,26 @@
+﻿//------------------------------------------------------------------------------
```
File can be kept, but with English header doc.

birkett commented on Properties/Settings.settings:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version='1.0' encoding='utf-8'?>
```
Not needed?

birkett commented on README.md:
```diff
@@ -2,22 +2,18 @@
-<br /><br />
 ## Requirements
 ####OS Support
 * Client: Windows XP / Vista / 7 / 8 / 8.1
```
See comment above regarding .NET support.

birkett commented on README.md:
```diff
@@ -2,22 +2,18 @@
+https://www.microsoft.com/en-us/download/details.aspx?id=55170
 <br /><br />
 ## Building
 Solution is known to build with Visual Studio 2013 (Community, Pro and Ultimate tested, Express should work), no additional dependencies are needed.<br />
```
See comment above regarding VS2013 support.

birkett commented on app.config:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Not needed?

birkett commented on packages.config:
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
See comment above regarding native alternatives.

birkett commented on packages/Newtonsoft.Json.11.0.2/LICENSE.md:
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
```
If this is required - I don't want it in the repository.
Please bring it in as a git submodule, or some other NuGet magic?

DaniGTA committed:
More clean up

DaniGTA committed:
Update was too fast

Just slow it down a bit

DaniGTA committed:
New Functions

# Right click on entry
 -Force Update

#Main Table
- Last Change
- Changing IP text

DaniGTA committed:
Fix bug

Fixed a bug with records saving

DaniGTA committed:
Use Usertime zone instead of UTC

DaniGTA committed:
Start on minimized bug fixed

DaniGTA committed:
Maybe a bug fix

try too fix a crash in the settings

DaniGTA committed:
Fix "Cant find new IP |System.ArgumentNullException"

DaniGTA committed:
Fix crash on wrong name and password.

DaniGTA committed:
GITIGNORE und GITATTRIBUTES hinzufügen.

DaniGTA committed:
Projektdateien hinzufügen.

DaniGTA committed:
Add Autostart

DaniGTA committed:
Fixed a bug with multi domains.

Commit:
Hide from ALT + TAB

DaniGTA committed:
Network Interface is now Selected able for get request

DaniGTA committed:
merged

DaniGTA committed:
fixed code

DaniGTA committed:
new default urls

DaniGTA committed:
Update CloudFalreAPI Auth

DaniGTA committed:
Update ReadMe

krisz-gemini committed:
FIX: Authorization errors

krisz-gemini committed:
FIX: ipify.org IP6 address

krisz-gemini committed:
FIX: update and failed counters weren't increased in the update records log message

DaniGTA committed:
Merge pull request #10 from krisz-gemini/master

Fix authorization errors and two other small bugs
